Deployment Guides/Provisioning-Islandora-on-Ubuntu-16.md

```diff
@@ -0,0 +1,1814 @@
++                <constructor-arg value="fedora://model#"/>
+                <constructor-arg value="ri" />
+        </bean>
+```
+
+Edit `$FEDORA_HOME/server/config/fedora.fcfg`:
+
+`cp $FEDORA_HOME/server/config/fedora.fcfg $FEDORA_HOME/server/config/fedora.fcfg.bak`
+
+Comment out the `<param name="datastore" value="localMulgaraTriplestore">` clause.
+
+Edit `$FEDORA_HOME/server/bin/env-server.sh`:
+
+`cp $FEDORA_HOME/server/bin/env-server.sh $FEDORA_HOME/server/bin/env-server.sh.bak`
+
+In the `execWithTheseArgs()` function:
+```
+-	-cp \"$webinf\"/classes:\"$FEDORA_HOME\"/server/bin:\"$webinf\"/lib/* \
++	-cp \"$webinf\"/classes:/usr/local/trippi-sail/*:\"$FEDORA_HOME\"/server/bin:\"$webinf\"/lib/* \
+```
+
+`systemctl stop blazegraph`
+
+Edit `/etc/bigdata/log4j.properties`:
+
+Set the default log level to `ERROR` instead of `WARN` (3 lines near the top).
+
+`systemctl start blazegraph`
+
+`systemctl restart tomcat7`
+
+#### GSearch and Solr <a id="gsearch-solr"></a>
+
+Note: part of the reason we still use older (now very old) versions of Solr is that, as of Solr 5.0 (latest is 7.3), installing Solr as a WAR servlet under containers like Tomcat is no longer supported. Also, quite significant changes in the configuration files have also happened, which makes use of the `basic-solr-config` package from Discovery Garden more difficult. DigiBESS has moved only from Solr 4.2.0 to 4.10.4 and already there are significant changes in `solrconfig.xml` and `schema.xml`. Since some will wish to stick with the tried and true, we include using 4.10.4 as optional. Hopefully this can be part of future migrations to more recent versions of Solr. Set the version of Solr you wish to use in your `install.properties`.
+
+Grab fedoragsearch and solr - do basic config:  
+```
+cd ~  
+
+git clone https://github.com/discoverygarden/gsearch.git
+
+cd gsearch/FedoraGenericSearch
+
+ant buildfromsource
+
+cd ~
+
+cp -v gsearch/FgsBuild/fromsource/fedoragsearch.war $CATALINA_HOME/webapps
+
+chown $FEDORA_USER:$FEDORA_USER $CATALINA_HOME/webapps/fedoragsearch.war
+
+rm -rf ~/gsearch
+
+wget $SOLR_URL
+
+tar -xf $SOLR_NAME.tgz
+```
+If you wish to install Solr under the same Tomcat as Fedora, do the following:
+```
+cp -r $SOLR_NAME/example/solr $FEDORA_HOME/solr
+
+cp $SOLR_NAME/example/webapps/solr.war $CATALINA_HOME/webapps/ && unzip -o $SOLR_NAME/example/webapps/solr.war -d $CATALINA_HOME/webapps/solr 
+
+mkdir $FEDORA_HOME/solr/$SOLR_DEFAULT_CORE_PATH/data
+
+chown -R $FEDORA_USER:$FEDORA_USER $FEDORA_HOME
+```
+If you wish to install Solr under its own container, do the following (from http://dev.digibess.it/doku.php?id=reloaded:be_solr). We use Jetty, rather than Tomcat, because that's what the Solr developers have chosen to use ("Only Jetty recieves testing... No tests are done with other containers before release..." (http://lucene.472066.n3.nabble.com/difference-between-apache-tomcat-vs-Jetty-td4096680.html)).
```

Comment:
In this setup we have 3 separate servlet containers.
-Fedora, Image server.. etc
-Solr
-Blazegraph

Wondering if we should combine Solr and Blazegraph in the same container for better memory utilization?

Reply:
@lutaylor @csqrs my 2 cents here.
I personally had better luck running Blazegraph and Solr as independent jetty services. (No tomcat). Give you much better control on initialization, server separation, logging, and most important, needed memory assignment. If you are all interested I can contribute with Solr 5.5.5 deployment guide and files, running on production for 7 months already with positive experience

Reply:
Also, DigiBESS puts Adore-Djatoka under the Blazegraph Tomcat, while I followed the older version of this document and put it under the Fedora Tomcat. I mention putting it under Blazegraph as an option. But I guess there's a move away from that to Cantaloupe anyway...

Reply:
@DiegoPino @csqrs 
It doesn't make a difference to me if we go with Jetty or Tomcat. I was just wondering if we would have better memory optimization on a  single server with fewer servlet containers if we are still using solr4x and other apps that ship as wars?.  Solr was still testing with Tomcat for Solr 4.x. For later versions, it is a stand-alone server https://wiki.apache.org/solr/SolrTomcat.

Maybe break things out when the app ships as a jar/stand alone and share the servlet container when deploying wars on the same server? 

@DiegoPino 
I think everyone would be interested in seeing your configurations!